bpf/init.sh

```diff
@@ -672,6 +681,18 @@ for iface in $(ip -o -a l | awk '{print $2}' | cut -d: -f1 | cut -d@ -f1 | grep
 	done
 done
 
+case "${MODE}" in
+	"flannel")
+		;;
+	"ipvlan")
+		;;
+	*)
+		COPTS=""
+		CALLS_MAP="cilium_calls_netdev_ns_${ID_HOST}"
+		bpf_load $NATIVE_DEVS "$COPTS" "ingress" bpf_netdev.c bpf_netdev.o from-netdev $CALLS_MAP
```

Comment:
I don't think adding a new BPF program is the right way to fix this. We already attach BPF programs to native devices from Golang (cf. [`pkg/datapath/loader/loader.go`](https://github.com/cilium/cilium/blob/c216dfb3ad911ca5ba7659b6266109c8c6882d70/pkg/datapath/loader/loader.go#L195)), so this will overwrite them. Doing so will also bypass the host firewall and break NodePort.

Instead, we should probably extend `icmp6_handle()`, which is already called from `bpf_host`:
https://github.com/cilium/cilium/blob/c216dfb3ad911ca5ba7659b6266109c8c6882d70/bpf/bpf_host.c#L370

Please see #12049 though. The `icmp6_handle()` call in `bpf_host` is currently broken (not compiled in) and the first commit in that pull request fixes it. The bug was introduced in v1.8 and we will backport that PR there.

Reply:
I noticed `bpf_host` is broken, too. glad that you are fixing it. I'll look into #12049 too. 

I also updated the `icmp6_handle()`. My intention is for the node to respond to the NS msg from external so I attached it on `native_dev`.  If I have questions, I'll ping you on slack.

Reply:
If you try #12049 out, it might address your concern. In its current state, it will fix `bpf_host` to let all ICMPv6 NS/NA messages up the stack (in accordance with RFC4890).

Reply:
I've tried #12049, the pods won't receive NS message, because the [destination IP](https://en.wikipedia.org/wiki/Solicited-node_multicast_address) is not the pod IP. I still need this but I will rebase it on top of #12049 . I will remove the new ebpf program and reuse `bpf_host`.